src/main/python/verification/text_verify.py

```diff
@@ -0,0 +1,155 @@
+                percentage match is above the specified threshold value, while the second return value is the total
+                percentage match value if verbose is False, or returns a dict of all the determined percentage match
+                values if verbose is True.
+        """
+        if type(extracted) is not dict:
+            raise TypeError('Bad type for arg extracted - expected dict. Received type ' + str(type(extracted)))
+        if type(verifier) is not dict:
+            raise TypeError('Bad type for arg verifier - expected dict. Received type ' + str(type(verifier)))
+        if type(threshold) is not float:
+            raise TypeError('Bad type for arg threshold - expected float. Received type ' + str(type(threshold)))
+        if type(min_matches) is not int:
+            raise TypeError('Bad type for arg min_matches - expected int. Received type ' + str(type(min_matches)))
+        if type(verbose) is not bool:
+            raise TypeError('Bad type for arg verbose - expected bool. Received type ' + str(type(verbose)))
+        # Set minimum number of matches, if zero or less set to one.
+        min_matches = min_matches if min_matches > 0 else 1
+        min_percentage = threshold * 100
+        # Logging for debugging and verbose purposes.
+        logger.debug('Threshold for verification set as: ' + str(min_percentage))
+        logger.debug('Minimum number of matches for verification set as: ' + str(min_matches))
+        logger.debug('Simplified scores to be returned' if not verbose else 'Verbose scores to be returned')
+        logger.debug('Verifying:')
+        # Prettify and log the extracted information.
+        [logger.debug(log_line) for log_line in prettify_json_message(extracted).split('\n')]
+        logger.debug('Against:')
+        # Prettify and log the verifier information.
+        [logger.debug(log_line) for log_line in prettify_json_message(verifier).split('\n')]
+        # Initialise a dictionary to house the final matching percentages.
+        scores = {}
+        # Iterate over the verifier and calculate a percentage match for the values,
+        # if the keys match and the corresponding values exist.
+        for key, value in verifier.items():
+            if key in extracted and extracted[key]:
+                logger.debug('Computing match "' + str(value) + '" and "' + str(extracted[key]) + '"...')
+                scores[key] = self._percentage_match(value, extracted[key])
+                logger.debug('"' + value + '" and "' + extracted[key] + '" match percentage is : ' + str(scores[key]))
+            else:
+                logger.warning('Could not find corresponding field "' + key + '" in extracted information to verify')
+        # Determine the number of scores calculated and initialise a default value for the total match score.
+        num_scores = len(scores)
+        total_score = 0.0
+        # Check if enough matches were found.
+        if num_scores >= min_matches:
```

Comment:
The way I see it is that if we do not receive enough extracted information for matching, then how can we definitively say that the person in question is 100% a match. For instance, if we only manage to get two fields, say Status and Surname, while the rest was not found, if said two extracted fields are both a 100% match then the overall match returned will also be 100%.